### ISSUE TITLE
Link CTRL+clicking, Shift+clicking, dragging, keyboard navigation in view mode may trigger twice or does the wrong thing

### ISSUE THREAD

**Issue by samueldr-at-cyberus** [outline/outline#13596]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### This is not related to configuring Outline

- [x] The issue is not related to self-hosting config

### Current Behavior

Following links in the "view" mode of outline is problematic in a couple of ways:

 - CTRL+clicking links may not follow the browser default behaviour
 - Shift+clicking links may not follow the browser default behaviour
 - Following links with keyboard navigation does not work
 - Dragging a link does not work

(More details in "steps to reproduce".)

I have not looked at the behaviour with "separate editing" turned off. I expect it may be weird in exciting new ways.

### Expected Behavior

The behaviour of links should not be changed compared to what the browser implements.

Open a page that has no SPA-specific behaviour, and check what the different outcomes are.

For example, https://www.example.com/ has such a well-behaved link.

### Steps To Reproduce

Make a new draft with the following content:

```
[this `is a` test](https://example.com)
```

Note that when typed manually the <code>\`is a\`</code> formatting may be eaten. The important part is that there **must** be a child element in the link. The child element may also be a highlight.

* * *

First, the base behaviour:

 - CTRL+click the link by clicking `this` or `test`
     - Firefox: link opens in a new tab
     - Chrome: link opens in a new tab
 - Shift+click the link by clicking `this` or `test`
     - Firefox: link opens in a new window
     - Chrome: link opens in a new **tab**

The behaviour of a child element:

 - CTRL+click the link by clicking `is a`
     - Firefox: link opens twice in new tabs
     - Chrome: link opens once in new tab
 - Shift+click the link by clicking `is a`
     - Firefox: link opens twice in two new windows
     - Chrome: link opens once in new **tab**

For keyboard navigation:

 - Use tab to provide the focus to the link
 - Use enter, CTRL+enter, Shift+enter
 - None of those work as they should (both Firefox and Chrome)

For dragging:

 - Try dragging the link (this could be used to make a bookmark in the bookmark toolbar).
 - *The dragging action is interrupted by a new tab opening*.
 - This also breaks the affordance of being able to cancel a misclick by dragging the link a bit.

### Environment

```markdown
- Outline: 1.9.2
- Browser: Firefox and Chrome

Observed on a Linux system, using Firefox, the behaviour differs in Chrome, which can hide the underlying issue, but is definitely also affected.
```

### Anything else?

The root cause is also what made #8725 necessary.

This is caused by the tortuous logic used in [`Link.tsx`](https://github.com/outline/outline/blob/d55e7f37e551a72d7b959ededa79e267fb946f1d/shared/editor/marks/Link.tsx#L186-L276).

First, `mousedown` is never the appropriate action for a link. A link is activated on click. This is why keyboard navigation and drag are broken.

  - The `mousedown` event never fires when using a keyboard
  - The `mousedown` event triggers even though a navigation was not yet requested when dragging

The misc. handling for link stuff in `mousedown` should be moved into the `click` handler, **but** care must be taken. It's not simply a matter of moving the code around, some of it is entirely unneeded, and tries to re-implement native behaviour that cannot be faithfully replicated by a website. The webapp should not try to hijack the navigation when a link is external to the webapp. It should leave the default behaviour continue. The late sanitization of links, though, can still be handled in `click`, and can still `preventDefault()` and `stopPropagation()`.

In addition, the weirdness with child element comes from how `mousedown` gets the closest `<a>`, while `click` does not. The assumption leads to a check where a `click` event on a link targets a child element, which is not an `HTMLAnchorElement`, and is the root cause of the other part of the issues.

```
mousedown: (view: EditorView, event: MouseEvent) => {
  const target = (event.target as HTMLElement)?.closest("a");
```

There is an accessibility component to this issue, as this is preventing keyboard navigation to work as designed. The breakage in the affordance for dragging a link is also problematic. I have also not verified what happens on touch-based events, but since `mousedown` is used, it may or may not have additional consequences for touch-based navigation regarding default behaviour.
